Disable "hello-world" workflow for (other) directories.

KsaR99 opened the pull request:
Currently It runs on any commit to any file.

PEZ commented:
It should run for changes to `compile.sh` and `run.sh`. Maybe better with a blacklist of directories. So not run for files in `loops`, `levenshtein`, or `fibonacci`.

KsaR99 commented:
- I added compile.sh & run.sh to white list.
- Also removed "components: 'native-image'" due to [issue in actions](https://github.com/bddicken/languages/actions/runs/12403096425)

![issue](https://github.com/user-attachments/assets/08fc35d9-c09e-4309-bc54-5afa15118399)



> Please remove "components: 'native-image'" from your workflow file. It is automatically included since GraalVM for JDK 17: https://github.com/oracle/graal/pull/5995

PEZ commented:
To me it makes total sense to do these changes, @bddicken. I don't think we'll be running the other benchmarks in CI anyway. At least not on all commits.
